backends/socket/socket_backend.c

```diff
@@ -184,6 +185,8 @@ int nanompi_free_socket_backend(nanompi_communicator_t *comm)
     int rank = comm->my_rank;
     int i;
 
+
+    free(mpi_poll_fd_init(comm, 0));
```

Comment:
I think this causes a memory leak. I tried to fix yesterday, but ended up breaking a lot and reverted. Gonna take another crack at this now.

Reply:
I think I resolved this in https://github.com/Quentin-Anthony/nanoMPI/pull/5/commits/3cdf68e8f922e4283293ad8d8d5aa1f50606950b

check my logic